### ISSUE TITLE
Act Runner registration fails with http code 500 

### ISSUE THREAD

**Issue by Mythos** [go-gitea/gitea#32029]
### Description

When trying to register my act runner, I'm getting a 500 error from the gitea instance.
In this case I wanted to re-register my runner with a different token, so I've removed the old one first and then regenerated the token.
After restarting the runner container, I'm getting those errors.

This is the log of the runner container:
```
time="2024-09-11T20:31:18Z" level=info msg="Starting runner daemon"
time="2024-09-11T20:31:18Z" level=error msg="fail to invoke Declare" error="unknown: rpc error: code = Unauthenticated desc = unregistered runner"
Error: unknown: rpc error: code = Unauthenticated desc = unregistered runner
time="2024-09-11T20:31:18Z" level=info msg="Starting runner daemon"
time="2024-09-11T20:31:18Z" level=error msg="fail to invoke Declare" error="unknown: rpc error: code = Unauthenticated desc = unregistered runner"
Error: unknown: rpc error: code = Unauthenticated desc = unregistered runner
time="2024-09-11T20:31:19Z" level=info msg="Starting runner daemon"
time="2024-09-11T20:31:19Z" level=error msg="fail to invoke Declare" error="unknown: rpc error: code = Unauthenticated desc = unregistered runner"
Error: unknown: rpc error: code = Unauthenticated desc = unregistered runner
time="2024-09-11T20:31:20Z" level=info msg="Starting runner daemon"
time="2024-09-11T20:31:20Z" level=error msg="fail to invoke Declare" error="unknown: rpc error: code = Unauthenticated desc = unregistered runner"
Error: unknown: rpc error: code = Unauthenticated desc = unregistered runner
time="2024-09-11T20:31:21Z" level=info msg="Starting runner daemon"
time="2024-09-11T20:31:21Z" level=error msg="fail to invoke Declare" error="unknown: rpc error: code = Unauthenticated desc = unregistered runner"
Error: unknown: rpc error: code = Unauthenticated desc = unregistered runner
time="2024-09-11T20:31:23Z" level=info msg="Starting runner daemon"
time="2024-09-11T20:31:23Z" level=error msg="fail to invoke Declare" error="unknown: rpc error: code = Unauthenticated desc = unregistered runner"
Error: unknown: rpc error: code = Unauthenticated desc = unregistered runner
```

and these are the logs of the gitea container:
```
2024/09/11 22:30:09 ...eb/routing/logger.go:102:func1() [I] router: completed POST /api/actions/runner.v1.RunnerService/Declare for [REDACTED]:0, 500 Internal Server Error in 1.6ms @ <autogenerated>:1(http.Handler.ServeHTTP-fm)
2024/09/11 22:31:09 ...eb/routing/logger.go:102:func1() [I] router: completed POST /api/actions/runner.v1.RunnerService/Declare for [REDACTED]:0, 500 Internal Server Error in 1.6ms @ <autogenerated>:1(http.Handler.ServeHTTP-fm)
2024/09/11 22:31:18 ...eb/routing/logger.go:102:func1() [I] router: completed POST /api/actions/runner.v1.RunnerService/Declare for [REDACTED]:0, 500 Internal Server Error in 1.6ms @ <autogenerated>:1(http.Handler.ServeHTTP-fm)
2024/09/11 22:31:18 ...eb/routing/logger.go:102:func1() [I] router: completed POST /api/actions/runner.v1.RunnerService/Declare for [REDACTED]:0, 500 Internal Server Error in 0.9ms @ <autogenerated>:1(http.Handler.ServeHTTP-fm)
2024/09/11 22:31:19 ...eb/routing/logger.go:102:func1() [I] router: completed POST /api/actions/runner.v1.RunnerService/Declare for [REDACTED]:0, 500 Internal Server Error in 1.0ms @ <autogenerated>:1(http.Handler.ServeHTTP-fm)
2024/09/11 22:31:20 ...eb/routing/logger.go:102:func1() [I] router: completed POST /api/actions/runner.v1.RunnerService/Declare for [REDACTED]:0, 500 Internal Server Error in 1.0ms @ <autogenerated>:1(http.Handler.ServeHTTP-fm)
2024/09/11 22:31:21 ...eb/routing/logger.go:102:func1() [I] router: completed POST /api/actions/runner.v1.RunnerService/Declare for [REDACTED]:0, 500 Internal Server Error in 1.4ms @ <autogenerated>:1(http.Handler.ServeHTTP-fm)
2024/09/11 22:31:23 ...eb/routing/logger.go:102:func1() [I] router: completed POST /api/actions/runner.v1.RunnerService/Declare for [REDACTED]:0, 500 Internal Server Error in 0.8ms @ <autogenerated>:1(http.Handler.ServeHTTP-fm)
2024/09/11 22:31:27 ...eb/routing/logger.go:102:func1() [I] router: completed POST /api/actions/runner.v1.RunnerService/Declare for [REDACTED]:0, 500 Internal Server Error in 1.6ms @ <autogenerated>:1(http.Handler.ServeHTTP-fm)
2024/09/11 22:31:34 ...eb/routing/logger.go:102:func1() [I] router: completed POST /api/actions/runner.v1.RunnerService/Declare for [REDACTED]:0, 500 Internal Server Error in 1.5ms @ <autogenerated>:1(http.Handler.ServeHTTP-fm)
2024/09/11 22:31:47 ...eb/routing/logger.go:102:func1() [I] router: completed POST /api/actions/runner.v1.RunnerService/Declare for [REDACTED]:0, 500 Internal Server Error in 1.7ms @ <autogenerated>:1(http.Handler.ServeHTTP-fm)
2024/09/11 22:32:13 ...eb/routing/logger.go:102:func1() [I] router: completed POST /api/actions/runner.v1.RunnerService/Declare for [REDACTED]:0, 500 Internal Server Error in 1.8ms @ <autogenerated>:1(http.Handler.ServeHTTP-fm)
```

### Gitea Version

1.22.2

### Can you reproduce the bug on the Gitea demo site?

No

### How are you running Gitea?

I'm using `gitea/gitea:1.22.2` and `gitea/act_runner:nightly` as docker containers.

### Database

MySQL/MariaDB

**Comment by lunny** [comment]
How did you register your runner? Have you followed https://docs.gitea.com/usage/actions/act-runner ?

**Comment by Mythos** [comment]
> How did you register your runner? Have you followed [docs.gitea.com/usage/actions/act-runner](https://docs.gitea.com/usage/actions/act-runner) ?

Hey there, I've set it up according to the docs:

```yaml
services:
  gitea-runner:
    image: gitea/act_runner:nightly
    mem_limit: 512M
    container_name: gitea-runner
    restart: unless-stopped
    environment:
      TZ: Europe/Berlin
      CONFIG_FILE: /config.yaml
      GITEA_INSTANCE_URL: "https://git.example.com"
      GITEA_RUNNER_REGISTRATION_TOKEN: ${GITEA_RUNNER_REGISTRATION_TOKEN}
      GITEA_RUNNER_NAME: "gitea-runner"
      GITEA_RUNNER_LABELS: ""
    volumes:
      - type: bind
        source: /var/www/gitea/gitea-runner/config.yaml
        target: /config.yaml
      - /var/www/gitea/gitea-runner/data:/data
      - /var/run/docker.sock:/var/run/docker.sock
    ports:
      - 32496:32496
```

I had a working runner before, I've just changed the `GITEA_RUNNER_REGISTRATION_TOKEN` in my `.env` file and now I'm getting the 500 errors on gitea's side.

**Comment by Mythos** [comment]
After doing a quick test with the runner executable for Linux, I found out that registering works completely fine.
When trying again, it told me a config already exists and if I want to overwrite it.

With that info I found a `.runner` file inside the persisted /data directory, after deleting it and restarting the container, the runner is registered again.

TL;DR: I didn't expect the docker environment variables to be persisted and not being used after successfully registering the runner.
